crates/attestation/src/crypto/crl.rs

```diff
@@ -0,0 +1,279 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::crypto::SubjectPublicKeyInfoExt;
+
+use anyhow::{bail, Result};
+use der::Encode;
+use x509::crl::CertificateList;
+use x509::PkiPath;
+
+pub trait PkiPathCRLCheck<'a> {
+    fn check_crl(&self, crls: &'a [CertificateList<'a>]) -> Result<()>;
+}
+
+impl<'a> PkiPathCRLCheck<'a> for PkiPath<'a> {
+    fn check_crl(&self, crls: &'a [CertificateList<'a>]) -> Result<()> {
+        let mut iter = self.iter().peekable();
+        loop {
+            let cert = iter.next();
+            let next = iter.peek();
+            if cert.is_none() || next.is_none() {
+                break;
+            }
+            let cert = cert.unwrap();
+            let next = next.unwrap();
+            for crl in crls {
+                // Don't bother with an empty CRL
+                if crl.tbs_cert_list.revoked_certificates.is_none() {
+                    continue;
+                }
+
+                let revocations = crl.tbs_cert_list.revoked_certificates.as_ref().unwrap();
```

Comment:
Let's not do anything with any of the CRL contents until we verified the signature.